### PR TITLE
Milestone 2: Add/remove and edit

### PR DESCRIPTION
**Pull request by yonas44**
- An add module to add new tasks to the list.
- A task removal module to remove a specific task.
- A module to remove all checked tasks.

Thank you for reviewing my PR!

### CODE REVIEW

File: src/modules/taskLoader.js [hunk 1]
```diff
@@ -1,14 +1,92 @@
...
+    checkBox.checked = task.bool;
+    task.index = index;
+
+    // Add event listener for checkbox
+    checkBox.addEventListener('change', () => {
+      tasks[index].bool = !task.bool;
+      if (task.bool === true) {
+        text.style.textDecoration = 'line-through';
+        form.classList.remove('on');
+        dots.className = 'bi bi-three-dots-vertical';
```
Comment (Nyame-Wolf) [inline src/modules/taskLoader.js:27]: I suggest you add a delete method for an individual task. Currently this three dots don't do anything. Its part of the [requirements](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m3_add_remove.md) having a delete method, not one that clears all completed, but one that removes individual items.
see this video

https://user-images.githubusercontent.com/48546992/199582810-ace80120-5aee-4b09-8a43-d90b5ca21319.mov

Below is the highlighted requirement:

Screenshot 2022-11-02 at 22 10 48

File: src/modules/taskLoader.js [hunk 2]
```diff
@@ -1,14 +1,92 @@
...
+    const text = document.createElement('input');
+    text.className = 'task';
+    text.value = task.string;
+    if (task.bool === true) text.style.textDecoration = 'line-through';
+
+    // Add event listener for each added form
+    form.addEventListener('submit', (event) => {
+      event.preventDefault();
+      text.blur();
+      taskEditor(text, index, tasks);
```
Comment (Nyame-Wolf) [inline src/modules/taskLoader.js:45]: Kindly note that as per the instructions your indexes should match the items. meaning if i have one `1` tem on the array, its index is `1` as per the instructions
Screenshot 2022-11-02 at 22 10 48

As per the screenshot attached below, your indexes start from `zero` `0`
Screenshot 2022-11-02 at 22 29 58

**Review by Nyame-Wolf — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle: 

Hi @yonas44  :wave: 

Your project page is looking fantastic so far :star_struck: ! 

However, there is still some room for improvement, that can help you have better code and make your project professional :sunglasses: 



## Highlights :sparkles:

- Great work using the right GitHub Flow :100: 
- Edit functionality implemented and working well
- Items saving to the local storage
- Items checking `true` or `false` upon checking the checkbox
- Commits are descriptive and meaningful :+1: 
- Project Documentation (PR, README) is professional :necktie: 

Congrats on your work so far! :clap:  

## Required Changes ♻️

There are some issues that you still need to work on to go to the next project, but you are almost there! 
_For more information, check the comments under the review :arrow_down:_

## Optional suggestions :ballot_box_with_check: 

_Every comment with the **[OPTIONAL]** prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can improve the quality of your code :100:_

Happy coding and keep it up! :raised_hands: 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
  Please, remember to tag me **@Nyame-Wolf**  in your question so I can receive the notification.

  **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------

  _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)

**Comment by yonas44** [comment]
https://user-images.githubusercontent.com/72081991/199760340-4012400f-a38f-48b3-918d-5b34d0b31334.mp4

**Comment by yonas44** [comment]
Hello @Nyame-Wolf, so I tried to attach a video showing how I was able to remove a specific task, which I took from the actual [minimalist](https://web.archive.org/web/20180320194056/http://www.getminimalist.com:80/) page itself, but I addressed to the other issue you mentioned and thank you for your feedback.🙏

**Review by roshan-bajgain — APPROVED** [review]
Hi @yonas44, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- No linter errors. ✔️
- GitHub flow is followed. ✔️
- Professional README file. ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**